kernel/src/syscall/open.rs

```diff
@@ -68,3 +75,49 @@ pub fn sys_creat(path_addr: Vaddr, mode: u16, ctx: &Context) -> Result<SyscallRe
         AccessMode::O_WRONLY as u32 | CreationFlags::O_CREAT.bits() | CreationFlags::O_TRUNC.bits();
     self::sys_openat(AT_FDCWD, path_addr, flags, mode, ctx)
 }
+
+fn do_open(
+    fs_resolver: &FsResolver,
+    path: &FsPath,
+    flags: u32,
+    mode: InodeMode,
+) -> Result<InodeHandle> {
+    let open_args = OpenArgs::from_flags_and_mode(flags, mode)?;
+
+    let lookup_res = if open_args.follow_tail_link() {
+        fs_resolver.lookup_unresolved(path)?
+    } else {
+        fs_resolver.lookup_unresolved_no_follow(path)?
+    };
+
+    let inode_handle = match lookup_res {
+        LookupResult::Resolved(target_path) => target_path.open(open_args)?,
+        LookupResult::AtParent(result) => {
+            if !open_args.creation_flags.contains(CreationFlags::O_CREAT) {
+                return_errno_with_message!(Errno::ENOENT, "file does not exist");
+            }
+            if open_args
+                .creation_flags
+                .contains(CreationFlags::O_DIRECTORY)
+            {
+                return_errno_with_message!(Errno::ENOTDIR, "cannot create directory");
+            }
+            if result.target_is_dir() {
+                return_errno_with_message!(Errno::EISDIR, "cannot create directory");
+            }
+
+            let (parent, tail_name) = result.into_parent_and_tail_name();
+
```

Comment:
Remove this blank line because `parent` and `tail_name` will only be used in the next line?